src_c/mixer.c

```diff
@@ -1348,8 +1348,15 @@ set_num_channels(PyObject *self, PyObject *args)
 
     MIXER_INIT_CHECK();
     if (numchans > numchanneldata) {
+        struct ChannelData *cd_org = channeldata;
         channeldata = (struct ChannelData *)realloc(
             channeldata, sizeof(struct ChannelData) * numchans);
+        if (!channeldata) {
+            /* Restore the original to avoid leaking it */
+            channeldata = cd_org;
+            PyErr_NoMemory();
```

Comment:
This could be just `return PyErr_NoMemory();`

Reply:
Thanks for the suggestion; I have updated the branch to implement it.